server/src/actors/project.rs

```diff
@@ -412,35 +411,125 @@ pub struct PublicKeyConfig {
 }
 
 /// Data for applying rate limits in Redis
+#[derive(Debug, Clone)]
+pub enum Quota {
+    RejectAll(RejectAllQuota),
+    Redis(RedisQuota),
+}
```

Comment:
The split here seems nice, but I'm wondering if this should rather be an implementation detail of the quotas component rather than part of the "public" interface?